.github/workflows/ci.yml

```diff
@@ -73,32 +33,76 @@ jobs:
+          else
+            echo "No Docker credentials, skipping login"
+          fi
+      - name: Pull Docker base images & warm Docker cache
         run: |
-          docker run --rm docker.pkg.github.com/${{ github.repository }}/dependabot-core-ci:${{ github.sha }} bash -c "cd /home/dependabot/dependabot-core/${{ matrix.suite }} && bundle exec rspec spec"
-
-  npm-and-yarn-js:
-    name: npm and yarn js
-    runs-on: ubuntu-latest
-    needs: ci-image
-    steps:
-      - name: Pull image from packages
+          docker pull ubuntu:18.04
+          docker pull "dependabot/dependabot-core-ci:core--$BRANCH_REF" ||
+            docker pull "dependabot/dependabot-core:latest"
+          docker pull "dependabot/dependabot-core-ci:ci--$BRANCH_REF" ||
+            "dependabot/dependabot-core-ci:latest" || true
+      - name: Build dependabot-core image
+        run: |
+          docker build \
+            -t "dependabot/dependabot-core:latest" \
+            -t "dependabot/dependabot-core-ci:core--$BRANCH_REF" \
+            --build-arg BUILDKIT_INLINE_CACHE=1 \
+            --cache-from ubuntu:18.04 \
+            --cache-from "dependabot/dependabot-core-ci:core--$BRANCH_REF" \
+            --cache-from "dependabot/dependabot-core:latest" \
+            .
+      - name: Push core image to packages
         run: |
-          docker login docker.pkg.github.com -u x -p ${{ secrets.GITHUB_TOKEN }}
-          docker pull "docker.pkg.github.com/${{ github.repository }}/dependabot-core-ci:${{ github.sha }}"
-          docker logout
-      - name: Run js linting
+          if [ -n "$DOCKER_USERNAME" ] && [ -n "$DOCKER_PASSWORD" ]; then
+            docker push "dependabot/dependabot-core-ci:core--$BRANCH_REF"
+          else
+            echo "No Docker credentials, skipping push"
+          fi
+      - name: Build dependabot-core-ci image
+        run: |
+          rm .dockerignore
+          docker build \
+            -t "dependabot/dependabot-core-ci:latest" \
+            -t "dependabot/dependabot-core-ci:ci--$BRANCH_REF" \
+            -f Dockerfile.ci \
+            --build-arg BUILDKIT_INLINE_CACHE=1 \
+            --cache-from ubuntu:18.04 \
+            --cache-from "dependabot/dependabot-core:latest" \
+            --cache-from "dependabot/dependabot-core-ci:ci--$BRANCH_REF" \
+            .
+      - name: Push ci image to packages
+        run: |
+          if [ -n "$DOCKER_USERNAME" ] && [ -n "$DOCKER_PASSWORD" ]; then
+            docker push "dependabot/dependabot-core-ci:latest"
+            docker push "dependabot/dependabot-core-ci:ci--$BRANCH_REF"
+          else
+            echo "No Docker credentials, skipping push"
+          fi
+      - name: Run Rubocop linting
+        run: |
+          docker run --rm dependabot/dependabot-core-ci bash -c "cd /home/dependabot/dependabot-core/${{ matrix.suite }} && bundle exec rubocop . -c ../.rubocop.yml"
```

Comment:
Aside: I was surprised to see this is a `matrix` build (e.g. that we're building+pushing these images so ~concurrently). 🏎️ .

Reply:
Ah yeah could push images only when on `common`

Reply:
or I guess docker will discard most the pushes as layer already exists?